src/query.rs

```diff
@@ -422,30 +422,42 @@ impl Query {
         Ok(blockhash)
     }
 
-    // Internal API for transaction retrieval (uses bitcoind)
-    fn _load_txn(&self, tx_hash: &Sha256dHash, block_height: u32) -> Result<Transaction> {
-        let blockhash = self.lookup_confirmed_blockhash(tx_hash, Some(block_height))?;
-        self.app.daemon().gettransaction(tx_hash, blockhash)
-    }
-
-    // Get transaction from txstore or the in-memory mempool Tracker
-    pub fn tx_get(&self, txid: &Sha256dHash) -> Option<Transaction> {
-        rawtxrow_by_txid(self.app.read_store(), txid)
-            .map(|row| deserialize(&row.rawtx).expect("cannot parse tx from txstore"))
-            .or_else(|| self.tracker.read().unwrap().get_txn(&txid))
+    // Load transaction by txid
+    pub fn load_txn(&self, txid: &Sha256dHash, blockhash: Option<&Sha256dHash>) -> Result<Transaction> {
+        if self.app.config().txstore_enabled {
+            // fetch from our txstore or mempool tracker
+            rawtxrow_by_txid(self.app.read_store(), txid)
+                .map(|row| deserialize(&row.rawtx).expect("cannot parse tx from txstore"))
+                .or_else(|| self.tracker.read().unwrap().get_txn(&txid))
+                .chain_err(|| format!("cannot find tx {}", txid))
+        } else {
+            // fetch from bitcoind
+            let blockhash = match blockhash {
+                Some(x) => Some(*x),
+                None => self.lookup_confirmed_blockhash(txid, None)?,
+            };
+            self.app.daemon().gettransaction(txid, blockhash.as_ref())
+        }
     }
 
-    // Get raw transaction from txstore or the in-memory mempool Tracker
-    pub fn tx_get_raw(&self, txid: &Sha256dHash) -> Option<Bytes> {
-        rawtxrow_by_txid(self.app.read_store(), txid)
-            .map(|row| row.rawtx)
-            .or_else(|| {
-                self.tracker
-                    .read()
-                    .unwrap()
-                    .get_txn(&txid)
-                    .map(|tx| serialize(&tx))
-            })
+    // Load raw transaction by txid
+    pub fn load_raw_txn(&self, txid: &Sha256dHash, blockhash: Option<&Sha256dHash>) -> Result<Bytes> {
```

Comment:
nit: may this method be implemented using `load_txn()`?

Reply:
It could, but it wouldn't be as efficient because it would need to deserialize and re-serialize the transaction, while the current `load_raw_txn()` implementation avoids deserialization in the first place.

Reply:
OK, sounds good.